text/3-entrees/fees.md

```diff
@@ -32,49 +30,123 @@ src:
 }
 ```
 
-## Converting Weight To Fees
+## 1 to 1 Conversion
 
-In many cases converting weight to fees in a one-to-one fashion, as shown above, will suffice and
-can be accomplished with
-[`ConvertInto`](https://crates.parity.io/sp_runtime/traits/struct.ConvertInto.html). This approach
-is also taken in the
-[node template](https://github.com/substrate-developer-hub/substrate-node-template/blob/43ee95347b6626580b1d9d554c3c8b77dc85bc01/runtime/src/lib.rs#L230).
+In many cases converting weight to fees one-to-one, as shown above, will suffice and can be
+accomplished with
+[`IdentityFee`](https://crates.parity.io/frame_support/weights/struct.IdentityFee.html). This
+approach is also taken in the
+[node template](https://github.com/paritytech/substrate/blob/2d39ec2c4aaec1cc0f91fcb91734de8f408dc1b2/bin/node-template/runtime/src/lib.rs#L246).
 It is also possible to provide a type that makes a more complex calculation. Any type that
-implements `Convert<Weight, Balance>` will suffice.
+implements
+[`WeightToFeePolynomial`](https://crates.parity.io/frame_support/weights/trait.WeightToFeePolynomial.html)
+will suffice.
 
-This example uses a quadratic conversion and supports custom coefficients.
+## Linear Conversion
 
-src:
-[`runtimes/weight-fee-runtime/src/lib.rs`](https://github.com/substrate-developer-hub/recipes/tree/master/runtimes/weight-fee-runtime/src/lib.rs)
+Another common way to convert weight to fees is linearly. When converting linearly, the weight is
+multiplied by a constant coefficient to determine the fee to charge. This is demonstrated in the
+`weight-fee-runtime` with the `LinearWeightToFee` struct.
+
+We declare the struct with a associated type `C` which will provide the coefficient.
 
 ```rust, ignore
-pub struct QuadraticWeightToFee<C0, C1, C2>(C0, C1, C2);
+pub struct LinearWeightToFee<C>(sp_std::marker::PhantomData<C>);
+```
 
-impl<C0, C1, C2> Convert<Weight, Balance> for QuadraticWeightToFee<C0, C1, C2>
-	where C0: Get<Balance>, C1: Get<Balance>, C2: Get<Balance> {
+Then we implement `WeightToFeePolynomial` for it. When implementing this trait, your main job is to
+return set of
+[`WeightToFeeCoefficient`](https://crates.parity.io/frame_support/weights/struct.WeightToFeeCoefficient.html)s.
+These coefficients can have integer and fractional parts and be positive or negative. In our
+`LinearWeightToFee` there is a single integer coefficient supplied by the associated type.
 
-	fn convert(w: Weight) -> Balance {
-		let c0 = C0::get();
-		let c1 = C1::get();
-		let c2 = C2::get();
-		let w = Balance::from(w);
+```rust, ignore
+impl<C> WeightToFeePolynomial for LinearWeightToFee<C>
+where
+	C: Get<Balance>,
+{
+	type Balance = Balance;
+
+	fn polynomial() -> WeightToFeeCoefficients<Self::Balance> {
+		let coefficient = WeightToFeeCoefficient {
+			coeff_integer: C::get(),
+			coeff_frac: Perbill::zero(),
+			negative: false,
+			degree: 1,
+		};
+
+		// Return a smallvec of coefficients. Order does not need to match degrees
```

Comment:
You have [a similar comment elsewhere in this PR](https://github.com/substrate-developer-hub/recipes/pull/269/files#diff-349b3dd3c5a6cfb0ad1b5850e089f95eR144), but it includes additional details. Do you want to update this one to match?

Reply:
Yeah I considered that too. Ultimately I decided it made more sense to introduce the consequences of negative coefficients in the `QuadraticWeightToFee` because that's where I actually use negative coefficients.

And if anyone wants to use `LinearWeightToFee` with a negative coefficient, they already have problems :)